lib/livebook/session.ex

```diff
@@ -948,6 +945,12 @@ defmodule Livebook.Session do
     Phoenix.PubSub.broadcast(Livebook.PubSub, "sessions:#{session_id}", message)
   end
 
+  defp notify_update(state) do
+    session = self_from_state(state)
+    Livebook.Sessions.update_session(session)
+    broadcast_message(state.session_id, {:session_updated, session})
```

Comment:
Do we still need this broadcast? Updating the session will trigger `handle_diff` which will do the relevant broadcasts when the information is available in the node.

Reply:
`handle_diff` sends updates on the `"sessions"` topic, while `SessionLive` only needs to subscribe to the specific `"session:{id}"` topic, where we send messages directly from the `Session` process.

Reply:
You could send the ID specific notifications from handle_diff too, no? I am just worried about sending updates we can’t yet read as they were not broadcast.

Reply:
What if we subscribe to `"sessions"` and match on the specific id for the update/closed events, or would that be too costy?

Reply:
That should be fine, as we don't expect sessions to be created that frequently. But if I understand correctly, this event is only used for LiveViews currently seeing that notebook? If so, shouldn't this be part of the "notebook feed" and be delivered directly to the session clients instead of broadcast? The same with removals, we can notify everyone in that notebook immediately.

Reply:
By broadcasting the updates we keep the homepage up-to-date as the notebook names or file paths change, or session are created/removed. At the moment we replicate the update/close events in the direct feed, but we could just use the `"sessions"` events for that.

Reply:
Right, I think it is important to not mix those. I think the home page should only read from the tracker feed, and the notebook only from the direct feed. It is important to not mix those, as we are doing here, otherwise this can happen:

1. The session is updated on the home page
2. The user refreshes the page and the update is no longer there

Or even we can receive events out of order. For example, an update for a session we don't even know it exists. So I think we should remove this broadcast here and keep it on the tracker. And we should always subscribe to one of them, not both, as otherwise we can't guarantee proper ordering.

Reply:
To be clear, the home page only reads the data from tracker and it subscribes to `"sessions"`, where only the tracker broadcasts.

The session page reads initial session info from the tracker as well. That's why it seems like a good idea to subscribe to `"sessions"` for those events and use `"session:{id}"` only for other direct messaging (so we would subscribe to both, each topic is a source of different events). How does this sound?

Reply:
I am still worried about ordering around all events. For example, does the update contain all events or only the changed fields? What happens if the session was removed after you started it? Maybe it is all correct at the end, the point is that I would prefer to not have to think about those. :)

So I would make it so the session page reads directly from the session process (since we need to connect to it anyway). Keep the tracker stuff only for the home page, and remove "session:#{id}"?